### ISSUE TITLE
Fails to fetch the correct gitlab url when publishing

### ISSUE THREAD

**Issue by jecoz** [goreleaser/goreleaser#1293]
**Describe the bug**
release failed after 3.01s error=GitHub/GitLab/Gitea Releases: failed to publish artifacts: POST https://api.github.com/repos/<etc>.

goreleaser is trying to publish to the `api.github.com` endpoint, the release was supposed to be published to a **private gitlab** server though!

**To Reproduce**

```yaml
project_name: sgccb
gitlab_urls:
  api: http://<host>/api/v4
  download: http://<host>
  # set to true if you use a self-signed certificate
  skip_tls_verify: true
env:
  - GO111MODULE=on

before:
  hooks:
    - go get -u --insecure ./...
builds:
  -
    binary: bin/{{ .ProjectName }}
    main: ./cmd/main.go
    goos:
      - darwin
      - linux
    goarch:
      - amd64
blobs:
  -
    provider: s3
    bucket: < bucket >
    region: < region >
    folder: "{{ .ProjectName }}/{{ .Version }}"
release:
  gitlab:
    owner: subgensdk
    name: "{{ .ProjectName }}"
  disable: false

signs:
  - artifacts: checksum
    args: ["--batch", "-u", "<my key>", "--output", "${signature}", "--detach-sign", "${artifact}"]
changelog:
  sort: asc

```

> Make sure your config file is valid by running
> `goreleaser check -f path-to-config-file`.
done!

**Expected behavior**
I expect goreleaser to try publishing the artifacts to the endpoints I provided  in the `gitlab_urls` section.

**Environment (please complete the following information):**

- OS: `mac`
- OS version: `Darwin jecoz.local 19.2.0 Darwin Kernel Version 19.2.0: Sat Nov  9 03:47:04 PST 2019; root:xnu-6153.61.1~20/RELEASE_X86_64 x86_64`
- GoReleaser Version: `version: 0.124.0
commit: 5f5f263aa8cc293a5684786af73ef68a7b77b902
built at: 2020-01-07T20:19:20Z
built by: goreleaser`
- GitLab version: `GitLab Community Edition 12.6.2`

**Additional context**
```
...
GITHUB/GITLAB/GITEA RELEASES
         • creating or updating release repo=subgensdk/sgccb tag=v0.1.5
         • release updated           url=
release failed after 3.01s ...
```
It looks like the `url` field is empty!

**Comment by jecoz** [comment]
Update 🤓: using `github_urls` instead of `gitlab_urls` updates the urls!
Still not the expected behaviour though, the `release` section is starts still with gitlab:
```
release:
  gitlab:
```

**Comment by caarlos0** [comment]
this seems like a bug indeed, will have a look soon.

feel free to try to open a pr if feel like it 👍

**Comment by jecoz** [comment]
Hi @caarlos0, I'll check it out too asap 😊

**Comment by caarlos0** [comment]
I tried to debug this, but didn't found out why this happens - yet...

**Comment by mavogel** [comment]
I could help too. As mentioned above it should be an expected behavior that the `gitlab_urls` are passed through the sections.

As a second step also not having to specify `gitlab` again in the `release` section, because it can be confusing it folks could mix up gitlab/github/gitea definitions throughout the config

**Comment by caarlos0** [comment]
If you could take a look that would be awesome @mavogel

**Comment by caarlos0** [comment]
not stale

**Comment by caarlos0** [comment]
are you setting the gitlab token somehow?

either with `GITHUB_TOKEN` env or `~/.config/goreleaser/github_token`?

I think this might be a side effect of #1389

**Comment by caarlos0** [comment]
FWIW I tried with a config like this:

```yaml
gitlab_urls:
  api: https://localhost:443/foo/bar
  download: https://localhost:443/downloads
release:
  gitlab:
    owner: foo
    name: bar
```

and then

```
GITLAB_TOKEN=adsasd goreleaser --rm-dist
```

**Comment by jecoz** [comment]
> either with GITLAB_TOKEN env or ~/.config/goreleaser/gitlab_token? 

I'm setting the token in `~/.config/goreleaser/gitlab_token`!

**Comment by caarlos0** [comment]
yeah, that was the issue then, should be fixed in #1389 

thanks!
